packages/loader/container-loader/src/protocolTreeDocumentStorageService.ts

```diff
@@ -0,0 +1,41 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    IDocumentStorageService,
+    ISummaryContext,
+} from "@fluidframework/driver-definitions";
+import {
+    ISummaryTree,
+} from "@fluidframework/protocol-definitions";
+
+export class ProtocolTreeStorageService implements IDocumentStorageService {
+    constructor(
+        private readonly internalStorageService: IDocumentStorageService,
+        private readonly generateProtocolTree: () => ISummaryTree,
+    ) {
+    }
+    public get policies() {
+        return this.internalStorageService.policies;
+    }
+    public get repositoryUrl() {
+        return this.internalStorageService.repositoryUrl;
+    }
+
+    getSnapshotTree = this.internalStorageService.getSnapshotTree.bind(this.internalStorageService);
+    getVersions = this.internalStorageService.getVersions.bind(this.internalStorageService);
+    write = this.internalStorageService.write.bind(this.internalStorageService);
+    createBlob = this.internalStorageService.createBlob.bind(this.internalStorageService);
+    readBlob = this.internalStorageService.readBlob.bind(this.internalStorageService);
+    downloadSummary = this.internalStorageService.downloadSummary.bind(this.internalStorageService);
+
+    async uploadSummaryWithContext(summary: ISummaryTree, context: ISummaryContext): Promise<string> {
+        summary.tree[".protocol"] = this.generateProtocolTree();
```

Comment:
@GaryWilber  @marcmasmsft will this cause any issue for odsp? I ran tests and did see and issues, everything seemed to round trip fine, but hard to know if it's behaving as expected, as scribe still overwrites the protocol tree

Reply:
Does it work (as is, in some definition of "works") for ODSP?

Reply:
I don't think this will work out. The summary will end up with 2 .protocol trees. There will be the .protocol tree that scribe adds, and the one that the client added in the ".app" tree. Today the server does not read the client uploaded summary, so it doesn't know that a .protocol tree is already there.

Reply:
That's what i'm afraid of. when we pull down the snapshot i only see single protocol tree, as i think it gets flattened, but i can't tell which on wins, as they are the same, but i assume odsp will store two copies. i guess the question is do we care? specifically, do we need to stage this, or wait for odsp to handle it?

Reply:
@GaryWilber does that mean ODSP creates a parent node (.app) for client uploaded summary? Which means this will translate as ".app/.protocol" in the storage? Who adds the .app? Driver, scribe, or ODSP rest api?

Reply:
Yea ODSP creates a parent ".app" node. This would translate as ".app/.protocol" in storage.